whipper/common/program.py

```diff
@@ -470,6 +470,47 @@ def getHTOA(self):
         stop = track.getIndex(1).absolute - 1
         return start, stop
 
+    def getCoverArt(self, path, cover_art, release_id):
+        """
+        Get cover art image from Cover Art Archive if user chooses to
+
+        :type  release_id: str
+        :param release_id: a release id (self.program.metadata.mbid)
+        :type  path: str
+        :param path: where to store the fetched image
+
+        :rtype: str or None
+        """
+        if not cover_art:
+            return None
+
+        import requests
+
+        image_path = os.path.join(path, 'coverart.jpg')
+        url = 'https://coverartarchive.org/release/%r/front' % release_id
+
+        logger.debug('fetching cover art at: %r', url)
+        try:
+            resp = requests.get(url, stream=True)
+        except requests.exceptions.ConnectionError as e:
+            logger.error('error fetching cover art: %r', e)
+            return None
+
+        if resp:
+            if resp.status_code == 200:
+                with open(image_path, 'wb') as f:
```

Comment:
The `content-type` check solution would be something like this (code to be added after line `if resp.status_code == 200:`):

```python
                allowed_img_fmts = {'image/jpeg', 'image/png'}
                if resp.headers['content-type'] not in allowed_img_fmts:
                    logger.warning("no cover art will be added because the "
                                   "fetched image is in an unsupported format")
                    return
```

Reply:
I think you can add this check too (leaving the other one in place) maybe with a comment explaining we check here (then in `encode.py` too) because this avoids downloading a file which won't be used. and this isn't the only check because the `content-type` header may be wrong.
Asking @Freso about this: could that header be incorrect? (or MusicBrainz always validates the mime type of the uploaded cover art files?)

Reply:
I don’t know if it could. I guess it could, but I don’t think so? If so, let’s cross that bridge when we get to it? Either way, MusicBrainz never even touches anything uploaded to the Cover Art Archive, it goes straight from the uploader’s machine to the IA’s machine(s). @MerlijnWajer would be a much better person to ask about what the IA does with received files… :)